### DISCUSSION TITLE
PYTHON support Build error

### DISCUSSION THREAD

**Discussion by SP-AWN** [IfcOpenShell/IfcOpenShell#5467] [Q&A] [answered]
Hi,
I'm trying to build IfcOpenshell from sources on Linux (Ubuntu 24.04)and I got such error.  Disabling BUILD_IFCPYTHON allows building process to finish with no errors.  SWIG version: 4.2.0.  I don't think I need PYTHON support  but still I'm curious what is the reason for error.
`
Built target cityjson_converter_exe
[100%] Swig compile IfcPython.i for python
[100%] Building CXX object examples/CMakeFiles/IfcAdvancedHouse.dir/IfcAdvancedHouse.cpp.o
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/ConversionResult.h:185: Warning 362: operator= ignored
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:175: Warning 362: operator= ignored
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:437: Warning 314: 'print' is a python keyword, renaming to '_print'

/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:437: Warning 314: 'print' is a python keyword, renaming to '_print'
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:455: Warning 314: 'print' is a python keyword, renaming to '_print'
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:455: Warning 314: 'print' is a python keyword, renaming to '_print'
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:1270: Warning 302: Redefinition of identifier 'cast' as ifcopenshell::geometry::taxonomy::cast ignored,
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:47: Warning 302: previous definition of 'cast' as ifcopenshell::geometry::taxonomy::cast.
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:1310: Warning 302: Redefinition of identifier 'dcast' as ifcopenshell::geometry::taxonomy::dcast ignored,
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/taxonomy.h:49: Warning 302: previous definition of 'dcast' as ifcopenshell::geometry::taxonomy::dcast.
/home/adam/projects/IfcOpenshell/src/ifcwrap/../serializers/XmlSerializer.h:38: Warning 325: Nested class not currently supported (Factory ignored)
/home/adam/projects/IfcOpenshell/src/ifcwrap/../ifcgeom/kernels/opencascade/IfcGeomTree.h:1005: Warning 325: Nested struct not currently supported (clash_task ignored)
swig4.0: ../../Source/CParse/templ.c:582: Swig_cparse_template_expand: Assertion `!SwigType_isvariadic(s)' failed.
`
Regards
Adam

**Comment by brunopostle** [comment]
These are just warnings, the problem is somewhere before this, search for 'error: ' in the output.

**Comment by aothms** [comment]
>  swig4.0: ../../Source/CParse/templ.c:582: Swig_cparse_template_expand: Assertion !SwigType_isvariadic(s)' failed. 

I do think it's this line. It appears to be something new in SWIG 4.2 as per the discussion https://github.com/swig/swig/issues/2794

It would be nice to know what `s` is in this assertion. I guess more people will start to run into this so this needs to be solved at some point.

**Reply by SP-AWN** [reply]
Thanks for the answer. I'm just preparing to build my own swig version from latest git. Because I think SWIG is the root of my other problems.  I will let you know if it helps
Regards
Adam

**Comment by SP-AWN** [comment, marked as answer]
OK. I have built FicOpneShell with Python support against my compilation of  SWIG V 4.3.0 (from github) No errors there. It seems that   orginal Ubuntu SWIG 4.2.0-2ubuntu1 package was the problem.
Regards
Adam
